components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/services/StorageProviderImpl.java

```diff
@@ -0,0 +1,173 @@
+     *
+     * @param taxonomyName String taxonomy name
+     */
+    @Override
+    public void removeTaxonomy(String taxonomyName) {
+        if (tenantTaxonomyMap != null) {
+            if (tenantTaxonomyMap.containsKey(tenantId)) {
+                tenantTaxonomyMap.get(tenantId).remove(taxonomyName);
+            }
+        }
+    }
+
+    /**
+     * This method will add taxonomy document bean object to the map for a given taxonomy name
+     *
+     * @param taxonomyBean Taxonomy document bean which contains taxonomy meta data
+     */
+    @Override
+    public void addTaxonomy(TaxonomyBean taxonomyBean) {
+        if (tenantTaxonomyMap != null) {
+            Map<String, TaxonomyBean> taxonomyMaps = tenantTaxonomyMap.get(tenantId);
+            if (taxonomyMaps != null) {
+                taxonomyMaps.put(taxonomyBean.getTaxonomyName(), taxonomyBean);
+                tenantTaxonomyMap.put(tenantId, taxonomyMaps);
+            } else {
+                Map<String, TaxonomyBean> tempTaxonomyMap = new HashMap<>();
+                tempTaxonomyMap.put(taxonomyBean.getTaxonomyName(), taxonomyBean);
+                tenantTaxonomyMap.put(tenantId, tempTaxonomyMap);
+            }
+        } else {
+            tenantTaxonomyMap = new HashMap<>();
+            Map<String, TaxonomyBean> tempTaxonomyMap = new HashMap<>();
+            tempTaxonomyMap.put(taxonomyBean.getTaxonomyName(), taxonomyBean);
+            tenantTaxonomyMap.put(tenantId, tempTaxonomyMap);
+        }
+    }
+
+    /**
+     * This method will update taxonomy document bean object in the map for a given taxonomy name
+     *
+     * @param oldName      String name of the existing name
+     * @param taxonomyBean Taxonomy document bean object
+     */
+    @Override
+    public void updateTaxonomy(String oldName, TaxonomyBean taxonomyBean) {
+        if (tenantTaxonomyMap != null) {
+            Map<String, TaxonomyBean> taxonomyMaps = tenantTaxonomyMap.get(tenantId);
+
+            if (taxonomyMaps.containsKey(oldName)) {
+                taxonomyMaps.remove(oldName);
+                taxonomyMaps.put(taxonomyBean.getTaxonomyName(), taxonomyBean);
+                tenantTaxonomyMap.put(tenantId, taxonomyMaps);
+
+            }
+
+        }
+    }
+
+    /**
+     * This method will initialize taxonomy maps with all taxonomy data.
+     *
+     * @throws UserStoreException           throws while getting RealmConfigurations
+     * @throws RegistryException            throws getting files from registry
+     * @throws IOException                  throws when reading file
+     * @throws SAXException                 throws when parsing content stream
+     * @throws ParserConfigurationException
+     */
+    @Override
+    public void initTaxonomyStorage()
+            throws UserStoreException, RegistryException, IOException, SAXException, ParserConfigurationException {
+        String TAXONOMY_PATH = GOVERNANCE_COMPONENT_PATH + "/taxonomy";
```

Comment:
Thank you Thushara,
I have fixed it with this PR https://github.com/wso2/carbon-governance/pull/253